src/WebJobs.Script/Description/DotNet/Compilation/CSharp/CSharpCompilation.cs

```diff
@@ -87,17 +88,17 @@ private static bool IsOrUsesAssemblyType(ITypeSymbol typeSymbol, IAssemblySymbol
                 && namedTypeSymbol.TypeArguments.Any(t => IsOrUsesAssemblyType(t, assemblySymbol));
         }
 
-        object ICompilation.Emit(CancellationToken cancellationToken) => Emit(cancellationToken);
+        async Task<object> ICompilation.EmitAsync(CancellationToken cancellationToken) => await EmitAsync(cancellationToken);
 
-        public Assembly Emit(CancellationToken cancellationToken)
+        public async Task<Assembly> EmitAsync(CancellationToken cancellationToken)
         {
             try
             {
                 using (var assemblyStream = new MemoryStream())
                 using (var pdbStream = new MemoryStream())
                 {
                     var compilationWithAnalyzers = _compilation.WithAnalyzers(GetAnalyzers());
-                    var diagnostics = compilationWithAnalyzers.GetAnalyzerDiagnosticsAsync().Result;
```

Comment:
This is a change I've been meaning to make for a while, but some issues with this blocking code are the primary driver.